RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,302 @@
+Validator Nodes and Base Nodes are Consensus Nodes(CN).
+Wallets and Token Wallets are Communication Clients(CC).
+CNs form the core communication infrastructure of the Tari communication network and are responsible for maintaining the Tari communication network by receiving, forwarding and distributing joining requests, discovery requests, data messages and routing information.
+CCs are different from CNs in that they do not maintain the network and they are not responsible for propagating any joining requests, discovery requests, data messages and routing information.
+They do make use of the network to submit their own joining requests and perform discovery request of other specific CNs and CCs when they need to communicate with them.
+Once a CC has discovered the CC or CN they want to communicate with, they will establish a direct P2P channel with them.
+The Tari communication network is unaware of this direct P2P communication once discovery is completed.
+
+The different entity types were grouped into the different communication node types as follows:
+
+| Entity Type    | Communication Node Type |
+|---             |---                      |
+| Validator Node | Consensus Node          |
+| Base Node      | Consensus Node          |
+| Wallet         | Consensus Client        |
+| Token Wallet   | Consensus Client        |
+
+#### Unique identification of Consensus Nodes and Consensus Clients 
+
+In the Tari communication network, each CN or CC makes use of a node id to determine their position in the network.
+This node id is either assigned based on registration on the Base Layer or can be derived from the CNs or CCs identification public key.
+The method used to obtain a node id will either enhance or limit the trustworthiness of that entity when propagating messages through them on the Tari communication network.
+When performing the broadcasting of data messages or propagating discovery requests, nodes with registration assigned node ids are considered more trustworthy compared to nodes with derived node ids. 
+
+Obtaining a node id from registration on the Base Layer is important as it will limit the potential of some parties performing [Eclipse attacks](https://eprint.iacr.org/2015/263.pdf) on the network.
+Registration makes it more difficult for [Bad Actor]s to position themselves in ideal patterns on the network to perform disruptive operations and actions. 
+In sensitive situations or situations where the Kademlia-style directed propagation of messages are vulnerable, gossip protocol-based broadcasting of messages can be performed as a less efficient, but safer alternative to ensure that the message will successfully reach the rest of the network.
+
+The recommended method of node id assignment for each Tari communication network entity type are as follows:
```

Comment:
I think this table should be added to the MUST requirement category.

Reply:
Changed to "MUST"